### DISCUSSION TITLE
Does dpc++ use clang plugin technology to support CUDA programming front-end?

### DISCUSSION THREAD

**Discussion by malixian** [intel/llvm#3328] [Q&A] [answered]
hipSYCL using clang plugin to understand SYCL code to hip or cuda, does dpc++ also use clang plugin technology to understand DPC++  to c++?
when running `clang -### -fsycl -fsycl-targets=nvptx64-nvidia-cuda-sycldevice -S -c -emit-llvm multi_device.cpp -o multi_device.ll` command , we can find `clang ... -x c++ test.o ... `. So I guess DPC++ frontedn transform dpc++ to c++.
Isn't right?

**Comment by alexbatashev** [comment, marked as answer]
@malixian no, the changes have been made permanently. See the initial PR for CUDA support in DPC++: https://github.com/intel/llvm/pull/1091

You may also find [this](https://intel.github.io/llvm-docs/CompilerAndRuntimeDesign.html) document interesting.
